examples/integrations/gcp/utils/resource_manager.py

```diff
@@ -0,0 +1,109 @@
+            table.title = t
+            table.align = "l"
+            for obj in objs:
+                row = []
+                for attr in attrs:
+                    # using attrgetter in order to get state.name for jobs
+                    row.append(operator.attrgetter(attr)(obj))
+                table.add_row(row)
+            print(table)
+
+
+def delete_object(object_types, object_id):
+    for t in object_types:
+        objs = getattr(aiplatform, t).list()
+        for obj in objs:
+            if obj.name == object_id:
+                print("DELETING " + obj.display_name)
+
+                # undeploy models from endpoints
+                if hasattr(obj, 'undeploy_all'):
+                    obj.undeploy_all()
+                obj.delete()
+
+
+def main():
+    parser = argparse.ArgumentParser(description='List and Delete Vertex '
+                                                 'AI objects')
+    parser.add_argument("--datasets", action="store_true", help='List datasets')
+    parser.add_argument("--models", action="store_true", help='List models')
+    parser.add_argument("--jobs", action="store_true", help='List jobs')
+    parser.add_argument("--endpoints", action="store_true", help='List endpoints')
+    parser.add_argument("--all", action="store_true", help='List all objects')
+    parser.add_argument("--delete", help='id of the object to delete. ')
+
+    # Print argparse help by default
+    if len(sys.argv) == 1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+
+    args = parser.parse_args()
+
+    print('You can also manage datasets, models, jobs and endpoints from the '
+          'GCP Console at')
+    print('https://console.cloud.google.com/vertex-ai\n')
+
+    if 'GOOGLE_APPLICATION_CREDENTIALS' not in os.environ.keys():
+        print('No application credentials found. Please make sure to set the '
+              'GOOGLE_APPLICATION_CREDENTIALS')
+        print('environment variable to a valid service account auth key.')
+        exit()
+    else:
+        print('Using auth credentials from ' +
+              os.environ['GOOGLE_APPLICATION_CREDENTIALS'])
+
+    dataset_types = ['ImageDataset', 'TabularDataset', 'TextDataset',
+                     'TimeSeriesDataset', 'VideoDataset']
+    dataset_attrs = ['name', 'display_name']
+
+    model_types = ['Model']
+    model_attrs = ['name', 'display_name']
+
+    job_types = ['AutoMLForecastingTrainingJob', 'AutoMLImageTrainingJob',
+                 'AutoMLTabularTrainingJob', 'AutoMLTextTrainingJob',
+                 'AutoMLVideoTrainingJob', 'BatchPredictionJob',
+                 'CustomContainerTrainingJob', 'CustomJob',
+                 'CustomPythonPackageTrainingJob', 'CustomTrainingJob',
+                 'HyperparameterTuningJob', 'PipelineJob']
+    job_attrs = ['name', 'display_name', 'state.name', 'has_failed']
+
+    endpoint_types = ['Endpoint']
+    endpoint_attrs = ['name', 'display_name']
```

Comment:
Is there no api that just allows users to see what resources are being used / incurring costs without specifying the type of resource? Even if we just provide like a `list_additional_resources` function that maybe just makes sure users don't expect 0 costs but they have something running.

Reply:
Apparently there isn't. I went through a little bit of a stackoverflow rabbit hole just now, ended up here:
https://issuetracker.google.com/issues/124508767?pli=1
It's a requested feature.

For our case, I think we need to assume the used has access to the console and will monitor costs from there. Also--as far as I can tell, the only objects my script lists that can incur costs are running jobs, and endpoints. Should I add some copy saying that, for extra clarity?

Reply:
Ok that is fine then. I think that this is a great start. If we feel like there are any gaps in this as we develop additional features we can always revisit it.